pkg/daemon/ceph/osd/volume.go

```diff
@@ -628,7 +631,7 @@ func (a *OsdAgent) initializeDevicesLVMMode(context *clusterd.Context, devices *
 					metadataDevices[md]["devices"] = deviceArg
 				}
 				deviceDBSizeMB := getDatabaseSize(a.storeConfig.DatabaseSizeMB, device.Config.DatabaseSizeMB)
-				if storeFlag == "--bluestore" && deviceDBSizeMB > 0 {
+				if (a.storeConfig.StoreType == string(cephv1.BlueStore) || a.storeConfig.StoreType == string(cephv1.BlueStoreRDR)) && deviceDBSizeMB > 0 {
```

Comment:
I think a helper could be good here. The string conversion makes this unnecessarily long. 

```suggestion
				if a.storeConfig.StoreType.IsAnyBluestoreType() && deviceDBSizeMB > 0 {
```

Reply:
done.